ui/plugin-system/src/runtime/datasources.ts

```diff
@@ -15,7 +15,13 @@ import { DatasourceSelector, DatasourceSpec } from '@perses-dev/core';
 import { createContext, useContext } from 'react';
 
 export interface DatasourceStore {
+  // TODO: Do we even need this method?
```

Comment:
So getDatasource could potentially be removed completely?

Reply:
Yeah, it feels like maybe you always just want the client, not the "config"? I may just remove it in the near future unless there's a use case for keeping it.